session.go

```diff
@@ -698,9 +702,22 @@ func (q *Query) defaultsFromSession() {
 	q.serialCons = s.cfg.SerialConsistency
 	q.defaultTimestamp = s.cfg.DefaultTimestamp
 	q.idempotent = s.cfg.DefaultIdempotence
+	q.metrics = initializeMetrics(s)
 	s.mu.RUnlock()
 }
 
+func initializeMetrics(s *Session) map[string]*queryMetric {
+	metrics := make(map[string]*queryMetric)
+	s.pool.mu.RLock()
+	defer s.pool.mu.RUnlock()
+
+	// Range over all defined IPs and prebuild metrics map
+	for _, hostConn := range s.pool.hostConnPools {
```

Comment:
Is there a possibility that a host is added to the pool after this is done? I'm pretty sure that can happen if you add a new node to the cluster and it's gossiped around the cluster through the control connection. In that case you might panic in the map accesses below.

Reply:
as mentioned above, will probably remove this completely.